crates/rome_js_analyze/src/semantic_analyzers/suspicious/no_duplicate_parameters.rs

```diff
@@ -201,5 +242,13 @@ fn track_binding(
 
 declare_node_union! {
     /// A union of all possible FunctionLike `JsAstNode` in the JS grammar.
-    pub(crate) AnyJsFunctionAndMethod = JsArrowFunctionExpression| JsFunctionDeclaration| JsFunctionExportDefaultDeclaration | JsFunctionExpression | JsMethodClassMember | JsMethodObjectMember
+    pub(crate) AnyJsFunctionAndMethod = JsArrowFunctionExpression| JsFunctionDeclaration| JsFunctionExportDefaultDeclaration | JsFunctionExpression | JsMethodClassMember | JsMethodObjectMember | JsConstructorClassMember
+}
+
+declare_node_union! {
+    pub(crate) AnyParameters = JsConstructorParameters | JsParameters
+}
+
+declare_node_union! {
+    pub(crate) AnyParameter = AnyJsConstructorParameter | AnyJsParameter
```

Comment:
I am wondering if this could be simplified to:

```suggestion
    pub(crate) AnyParameter = AnyJsFormalParameter | JsRestParameter | TsThisParameter | TsPropertyParameter
```

This could simplify `traverse_parameter`?